src/mbedtls/pk.pyx

```diff
@@ -990,6 +990,13 @@ cdef class ECDHBase:
     def _peer_public_key(self, ECPoint ecp):
         check_error(_pk.mbedtls_ecp_copy(&self._ctx.Qp, &ecp._ctx))
 
+    def _generate_public(self):
```

Comment:
Could you move this below the ```generate_secret()``` method and rename it ```generate_public_key()```? Also, the docstring could juts be ```"""Generate public key from a private key."""```

Reply:
My considering is that this is a little hackish. I think mbedtls should provide the method but I found none. This method should not be used in "common" scenarios. But due to cython's nature, this had to be added to be usable.
If you consider this acceptable, I'll do the change as you said.

Reply:
Yes, please apply the requested changes. The only goal of moving and renaming the method is so that I can find it later. The suggested doctstring fixes a few typos and conforms to PEP 257.

Reply:
@stepheny I am very happy with your pull request if you make that changes I have requested. After second thoughts, making everything public is a *bad* idea. In any case, I would favor immutable objects. Anyway.

Would you mind making the changes now? Otherwise, I will merge your PR and apply the patches myself. I am not sure how it looks in the git history—and how important it is to you.